python/tchannel/messages/call_response.py

```diff
@@ -27,3 +35,95 @@ class CallResponseMessage(CallRequestMessage):
         'arg_2',
         'arg_3',
     )
+
+    FLAGS_SIZE = 1
+    CODE_SIZE = 1
+    # sizeOf(span_id) + sizeOf(parent_id) + sizeOf(trace_id) = 3*8 = 24
+    TRACE_SIZE = 8
+    TRACEFLAGS_SIZE = 1
+    NH_SIZE = 1
+    HEADER_SIZE = 1
+    CSUMTYPE_SIZE = 1
+    CSUM_SIZE = CallRequestMessage.CHECKSUM[CSUMTYPE_SIZE]
+
+    ARG_LENGTH = 2
+
+    def parse(self, payload, size):
+        """Parse a call request message from a payload."""
+        self.flags = read_number(payload, self.FLAGS_SIZE)
+        self.code = read_number(payload, self.CODE_SIZE)
+
+        self.span_id = read_number(payload, self.TRACE_SIZE)
+        self.parent_id = read_number(payload, self.TRACE_SIZE)
+        self.trace_id = read_number(payload, self.TRACE_SIZE)
+
+        self.traceflags = read_number(payload, self.TRACEFLAGS_SIZE)
+
+        self.headers, _ = self._read_headers(
+            payload,
+            self.NH_SIZE,
+            self.HEADER_SIZE,
+        )
+
+        self.checksum_type = read_number(payload, self.CSUMTYPE_SIZE)
+        if self.checksum_type:
+            self.checksum = read_number(payload, self.CSUM_SIZE)
+
+        self.arg_1, _ = read_variable_length_key(
+            payload,
+            self.ARG_LENGTH,
+            decode=False,
+        )
+        self.arg_2, _ = read_variable_length_key(
+            payload,
+            self.ARG_LENGTH,
+            decode=False,
+        )
+        self.arg_3, _ = read_variable_length_key(
+            payload,
+            self.ARG_LENGTH,
+            decode=False,
+        )
+
+        cur = payload.tell()
+        payload.seek(0, os.SEEK_END)
+        end = payload.tell()
+
+        if cur != end:
+            logging.error("Extra space exists in the end of payload!")
+
+
+    def serialize(self, out):
+        """Write a call request message out to a buffer."""
+        out.extend(write_number(self.flags, self.FLAGS_SIZE))
+        out.extend(write_number(self.code, self.code_SIZE))
+
+        out.extend(write_number(self.span_id, self.TRACE_SIZE))
+        out.extend(write_number(self.parent_id, self.TRACE_SIZE))
+        out.extend(write_number(self.trace_id, self.TRACE_SIZE))
+        out.extend(write_number(self.traceflags, self.TRACEFLAGS_SIZE))
```

Comment:
Same response: extract code that is performing the exact same calls into a re-usable method.